engines/dfc_provider/app/services/supplied_product_builder.rb

```diff
@@ -90,10 +90,11 @@ def self.product_type(variant)
   end
 
   def self.taxon(supplied_product)
-    return unless supplied_product.productType
+    dfc_id = supplied_product.productType&.semanticId
 
-    dfc_id = supplied_product.productType.semanticId
-    Spree::Taxon.find_by(dfc_id: )
+    # Every product needs a primary taxon to be valid. So if we don't have
+    # one or can't find it we just take a random one.
+    Spree::Taxon.find_by(dfc_id:) || Spree::Taxon.first
```

Comment:
I trust that you've fully considered this, but it still raises flags. What if the wrong category is not noticed and persisted for a long time? 
Maybe if there were a catch-all, eg do we support the top-level "product" category? Probably not (and I recall you mentioning last week we only want a flat level anyway).

So yeah this will do, but I can imagine causing confusion in the future so it might be worth documenting somewhere, or notifying #product-circle.

Reply:
A default category would be good. I reckon that there will be a few changes around this. Once we have the proper DFC mapping on all instances, we can talk about standardising the categories on all instances and use the DFC list instead of our own database. Then we don't actually need to match anything any more. We could also talk about not requiring a category on products. I think that it came from Spree and we don't need to enforce it.